lib/datadog/open_feature/binding/configuration.rb

```diff
@@ -0,0 +1,292 @@
+        end
+      end
+
+      # Represents a variation value for a flag
+      class Variation
+        attr_reader :key, :value
+
+        def initialize(key:, value:)
+          @key = key
+          @value = value
+        end
+
+        def self.from_json(variation_data)
+          new(
+            key: variation_data['key'],
+            value: variation_data['value']
+          )
+        end
+      end
+
+      # Represents an allocation rule with traffic splits
+      class Allocation
+        attr_reader :key, :rules, :start_at, :end_at, :splits, :do_log
+
+        def initialize(key:, rules: nil, start_at: nil, end_at: nil, splits:, do_log: true)
+          @key = key
+          @rules = rules
+          @start_at = start_at
+          @end_at = end_at
+          @splits = splits || []
+          @do_log = do_log
+        end
+
+        def self.from_json(allocation_data)
+          new(
+            key: allocation_data['key'],
+            rules: parse_rules(allocation_data['rules']),
+            start_at: parse_timestamp(allocation_data['startAt']),
+            end_at: parse_timestamp(allocation_data['endAt']),
+            splits: parse_splits(allocation_data['splits'] || []),
+            do_log: allocation_data.fetch('doLog', true)
+          )
+        end
+
+        private
+
+        def self.parse_rules(rules_data)
+          return nil if rules_data.nil? || rules_data.empty?
+
+          rules_data.map { |rule_data| Rule.from_json(rule_data) }
+        end
+
+        def self.parse_splits(splits_data)
+          splits_data.map { |split_data| Split.from_json(split_data) }
+        end
+
+        def self.parse_timestamp(timestamp_data)
+          return nil if timestamp_data.nil?
+
+          # Handle both Unix timestamps and ISO8601 strings
+          case timestamp_data
+          when Numeric
+            Time.at(timestamp_data)
+          when String
+            Time.parse(timestamp_data)
+          else
+            nil
+          end
+        rescue StandardError
+          nil
+        end
+      end
+
+      # Represents a traffic split within an allocation
+      class Split
+        attr_reader :shards, :variation_key, :extra_logging
+
+        def initialize(shards:, variation_key:, extra_logging: nil)
+          @shards = shards || []
+          @variation_key = variation_key
+          @extra_logging = extra_logging || {}
+        end
+
+        def self.from_json(split_data)
+          new(
+            shards: parse_shards(split_data['shards'] || []),
+            variation_key: split_data['variationKey'],
+            extra_logging: split_data['extraLogging'] || {}
+          )
+        end
+
+        private
+
+        def self.parse_shards(shards_data)
+          shards_data.map { |shard_data| Shard.from_json(shard_data) }
+        end
+      end
+
+      # Represents a shard configuration for traffic splitting
+      class Shard
```

Comment:
I am not sure about this name - it's called `Shard`, but has a property called `total_shards`?